lib/hawk/model/association.rb

```diff
@@ -230,7 +231,7 @@ def #{entities}
                 return @_#{entities} if instance_variable_defined?('@_#{entities}')
                 params = clean_inherited_params(self.params, #{conditions})
 
-                @_#{entities} = #{parent}::#{klass}.where(params)
+                @_#{entities} = #{klass_name}.where(params)
```

Comment:
This is getting really :spaghetti: .. these `eval` blocks need refactoring in separate reusable micro-methods.